modules/internal/ChapelLocale.chpl

```diff
@@ -247,6 +247,35 @@ module ChapelLocale {
       return "";
     }
 
+    //
+    // Support for different types of memory:
+    // large, random-access (low latency), and streaming (high bandwidth)
+    // Inspired by madvise()'s support for RANDOM and SEQUENTIAL access
+    //
+    pragma "no doc"
+    proc defaultMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc largeMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc randomMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc streamingMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
```

Comment:
Answering "would we give them DDR because latency was first priority", it would depend on whether we decided the selector function should treat the characteristics as being ordered in terms of desirability.  If yes then yes, I think we'd want to give them DDR.  If no then definitely MCDRAM would be a possible answer (and if they wanted to skew us toward giving them DDR instead they could throw "large" into the set of desired characteristics).  In any case, I think this is something we could take offline from this PR.  A change to the style of selector I'm talking about seems too large for this PR now, both philosophically and in terms of getting it in well ahead of the release.  So let's go with what you have, modulo the change from program to hardware characteristics unless anyone objects to that.